enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/SlotAllocation.scala

```diff
@@ -271,12 +271,17 @@ object SlotAllocation {
         rhsPipeline
 
       case _:CartesianProduct =>
-        val cartesianProductPipeline = lhsPipeline.seedClone()
+        val newPipeline = lhsPipeline.seedClone()
         rhsPipeline.foreachSlot {
           case (k, slot) =>
-            cartesianProductPipeline.add(k, slot)
+            newPipeline.add(k, slot)
         }
-        cartesianProductPipeline
+        newPipeline
+
+      case RollUpApply(_, _, collectionName, _, _) =>
+        val newPipeline = lhsPipeline.seedClone()
```

Comment:
I don't think we need to do a `seedClone()` here. Pretty sure we should just add one reference slot to the `lhs` pipeline. For each incoming row, we're producing exactly one row, and the only difference between the incoming and outgoing row is that the outgoing one has one additional refslot, nothing else.

Reply:
yes, l think `lhs` should continue as well.